src/Core/Services/RequestValidator.cs

```diff
@@ -325,9 +322,11 @@ public static void ValidatePrimaryKeyRouteAndQueryStringInURL(EntityActionOperat
                     unvalidatedFields.Remove(exposedName!);
                 }
             }
-            // TO DO: If the request header contains x-ms-must-match custom header with value of "ignore"
-            // this should not throw any error. Tracked by issue #158.
-            if (unvalidatedFields.Any())
+
+            // There may be unvalidated fields remaining because of extraneous fields in request body
+            // which are not mapped to the table. We throw an exception only when we operate in strict mode,
+            // i.e. when extraneous fields are not allowed.
+            if (unvalidatedFields.Any() && IsRequestBodyStrict())
```

Comment:
Is there any scenario where `unvalidatedFields` could still contain a field that is supposed to have been validated and removed but is not? 

When running in non-strict mode, exceptions would not get thrown for those fields.

Reply:
Exactly what you described is happening here. If there are any unvalidated fields, we throw an exception only when we operate in strict mode. No exceptions would be thrown in non-strict mode.